spec/controllers/payments_controller_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe PaymentsController, type: :controller do
+  let(:event) { create(:event, max_adult_tickets_per_request: 1) }
+  let(:user) { create(:user) }
+  let(:ticket_request) { create(:ticket_request, event:, user:, kids: 0) }
+  let(:payment) { create(:payment, status: Payment::STATUS_IN_PROGRESS, stripe_payment_id: 'pi_3PF4524ofUrW5ZY40NeGThOz', ticket_request:) }
+
+  before { sign_in user if user }
+
+  describe 'POST #confirm' do
+    subject { post :confirm, params: { id: payment.id, event_id: ticket_request.event.id, ticket_request_id: ticket_request.id } }
+
+    context 'when payment status in progress' do
+      it { is_expected.to have_http_status(:ok) }
+    end
+
+    context 'when payment status new' do
+      let(:payment) { create(:payment, status: Payment::STATUS_NEW, stripe_payment_id: 'pi_3PF4524ofUrW5ZY40NeGThOz', ticket_request:) }
```

Comment:
Let's move the value of `pi_3PF4524ofUrW5ZY40NeGThOz` into `let(:stripe_payment_id)`